rspace/src/main/scala/coop/rchain/rspace/HotStore.scala

```diff
@@ -250,20 +260,23 @@ private class InMemHotStore[F[_]: Sync, C, P, A, K](
         .toVector
     } yield (continuations ++ data ++ joins)
 
-  private def removeIndex[E](col: Seq[E], index: Int): F[Seq[E]] =
-    if (col.isDefinedAt(index)) {
-      val (l1, l2) = col splitAt index
-      (l1 ++ (l2 tail)).pure[F]
-    } else
+  private def removeIndex[E](col: Seq[E], index: Int): Seq[E] = {
+    val (l1, l2) = col splitAt index
+    (l1 ++ (l2 tail))
+  }
+
+  private def checkIndex[E](col: Seq[E], index: Int): F[Unit] =
+    if (!col.isDefinedAt(index)) {
```

Comment:
I use the Applicative[F].whenA or unlessA pattern - I find it more readable and terser. But YMMV. So this could be rewritten as:
```scala
Applicative[F].unlessA(col.isSefinedAt(index))(Sync[F].raiseError(...))
```